### PR TITLE
add ec2 instance type to build_start_url payload

### PR DESCRIPTION
**Pull request by matthewalbani**

### CODE REVIEW

File: lib/kochiku/settings.rb [hunk 1]
```diff
@@ -18,6 +18,7 @@ def initialize(root_dir)
         contents["git_strategy"] = user_defined_options["git_strategy"] || "localcache"
         contents["git_shared_root"] = user_defined_options["git_shared_root"]
         contents["logstreamer_port"] = user_defined_options['logstreamer_port']
+        contents["in_ec2"] = user_defined_options['in_ec2'] || false
```
Comment (robolson) [inline lib/kochiku/settings.rb]: I am not a fan of `in_ec2`. Do you like any of these names better?
* running_on_ec2
* is_ec2_worker
* inside_ec2

File: lib/kochiku/worker.rb [hunk 2]
```diff
@@ -60,6 +60,32 @@ def msg2str(msg)
       def build_strategy
         @build_strategy ||= BuildStrategyFactory.get_strategy(settings.build_strategy)
       end
+
+      def instance_type
+        @instance_type ||= begin
+          path = File.join(ENV["HOME"], 'instance_type.txt')
```
Comment (robolson) [inline lib/kochiku/worker.rb]: I don't like dumping stuff in `$HOME`, we should keep it inside of the kochiku-worker dir. Either kochiku-worker root or `kochiku-worker/tmp/`

File: lib/kochiku/worker.rb [hunk 3]
```diff
@@ -60,6 +60,32 @@ def msg2str(msg)
...
+          if File.file?(path)
+            instance_type = File.read(path)
```
Comment (robolson) [inline lib/kochiku/worker.rb]: does this need `.chomp`?

File: lib/kochiku/worker.rb [hunk 4]
```diff
@@ -60,6 +60,32 @@ def msg2str(msg)
...
+          instance_type
+        end
+      end
+
+      def request_instance_type_from_ec2
+        begin
+          instance_type = RestClient::Request.execute(method: :get,
+                                                      url: "http://169.254.169.254/latest/meta-data/instance-type",
+                                                      timeout: 15,
+                                                      open_timeout: 15).body
```
Comment (robolson) [inline lib/kochiku/worker.rb]: 15 seconds is excessively long. 5 seconds sounded good. Additionally we only need to specify `open_timeout`. No way will reading this little bit of text take so long that we need a custom `timeout`.

File: lib/kochiku/worker.rb [hunk 5]
```diff
@@ -60,6 +60,32 @@ def msg2str(msg)
...
+        rescue Exception => e
```
Comment (robolson) [inline lib/kochiku/worker.rb]: rescue the relevant RestClient exception only

File: lib/kochiku/settings.rb (hunk 1 above)
Comment (matthewalbani) [reply lib/kochiku/settings.rb]: i vote for running_on_ec2

File: lib/kochiku/worker.rb [hunk 6]
```diff
@@ -60,6 +60,31 @@ def msg2str(msg)
       def build_strategy
         @build_strategy ||= BuildStrategyFactory.get_strategy(settings.build_strategy)
       end
+
+      def instance_type
+        @instance_type ||= begin
+          path = File.join(File.dirname(__FILE__),'..', '..', 'tmp', 'instance_type.txt')
```
Comment (robolson) [inline lib/kochiku/worker.rb]: missing space

**Pull request merged**